diff -crB avro-cpp-1.7.5/CMakeLists.txt avro-cpp-1.7.5_patched/CMakeLists.txt
*** avro-cpp-1.7.5/CMakeLists.txt	2013-08-19 21:12:49.000000000 +0300
***************
*** 93,168 ****
  
  target_link_libraries (avrocpp ${Boost_LIBRARIES})
  
- add_executable (precompile test/precompile.cc)
- 
- target_link_libraries (precompile avrocpp_s ${Boost_LIBRARIES})
- 
- macro (gencpp file ns)
-     add_custom_command (OUTPUT ${ns}.hh
-         COMMAND precompile ${CMAKE_CURRENT_SOURCE_DIR}/jsonschemas/${file}
-             ${file}
-         COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/scripts/gen-cppcode.py
-             -n ${ns} -i ${file} -o ${ns}.hh
-         DEPENDS precompile ${CMAKE_CURRENT_SOURCE_DIR}/jsonschemas/${file})
-     add_custom_target(${ns} DEPENDS ${ns}.hh)
- endmacro (gencpp)
- 
- if (CYGWIN OR NOT WIN32)
-     gencpp (bigrecord testgen)
-     gencpp (bigrecord2 testgen2)
- endif ()
- 
- macro (gen file ns)
-     add_custom_command (OUTPUT ${file}.hh
-         COMMAND avrogencpp
-             -p -
-             -i ${CMAKE_CURRENT_SOURCE_DIR}/jsonschemas/${file}
-             -o ${file}.hh -n ${ns} -U
-         DEPENDS avrogencpp ${CMAKE_CURRENT_SOURCE_DIR}/jsonschemas/${file})
-     add_custom_target (${file}_hh DEPENDS ${file}.hh)
- endmacro (gen)
- 
- gen (bigrecord testgen)
- gen (bigrecord2 testgen2)
- gen (tweet testgen3)
- gen (union_array_union uau)
- gen (union_map_union umu)
- gen (union_conflict uc)
- gen (recursive rec)
- gen (reuse ru)
- gen (circulardep cd)
- 
- add_executable (avrogencpp impl/avrogencpp.cc)
- target_link_libraries (avrogencpp avrocpp_s ${Boost_LIBRARIES})
- 
- enable_testing()
- 
- macro (unittest name)
-     add_executable (${name} test/${name}.cc)
-     target_link_libraries (${name} avrocpp ${Boost_LIBRARIES})
-     add_test (NAME ${name} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-         COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name})
- endmacro (unittest)
- 
- unittest (buffertest)
- unittest (unittest)
- unittest (SchemaTests)
- unittest (CodecTests)
- unittest (StreamTests)
- unittest (SpecificTests)
- unittest (DataFileTests)
- unittest (JsonTests)
- unittest (AvrogencppTests)
- 
- if (CYGWIN OR NOT WIN32)
-     unittest (testgentest)
-     add_dependencies (testgentest testgen testgen2)
- endif()
- 
- add_dependencies (AvrogencppTests bigrecord_hh bigrecord2_hh tweet_hh
-     union_array_union_hh union_map_union_hh union_conflict_hh
-     recursive_hh reuse_hh circulardep_hh)
- 
  include (InstallRequiredSystemLibraries)
  
  set (CPACK_PACKAGE_FILE_NAME "avrocpp-${AVRO_VERSION_MAJOR}")
--- 93,98 ----
***************
*** 174,181 ****
      ARCHIVE DESTINATION lib
      RUNTIME DESTINATION lib)
  
- install (TARGETS avrogencpp RUNTIME DESTINATION bin)
- 
  install (DIRECTORY api/ DESTINATION include/avro
      FILES_MATCHING PATTERN *.hh)
  
--- 104,109 ----
